rust_snuba/rust_arroyo/src/processing/dlq.rs

```diff
@@ -121,6 +126,84 @@ impl<TPayload> DlqPolicy<TPayload> {
     }
 }
 
+// Wraps the DLQ policy and keeps track of messages pending produce/commit.
+type Futures<TPayload> = VecDeque<(u64, JoinHandle<BrokerMessage<TPayload>>)>;
+
+pub(crate) struct DlqPolicyWrapper<TPayload> {
+    dlq_policy: Option<DlqPolicy<TPayload>>,
+    runtime: Handle,
+    // This is a per-partition max
+    max_pending_futures: usize,
+    futures: BTreeMap<Partition, Futures<TPayload>>,
+}
+
+impl<TPayload: Clone + Send + Sync + 'static> DlqPolicyWrapper<TPayload> {
+    pub fn new(dlq_policy: Option<DlqPolicy<TPayload>>) -> Self {
+        let concurrency_config = ConcurrencyConfig::new(10);
+        DlqPolicyWrapper {
+            dlq_policy,
+            runtime: concurrency_config.handle(),
```

Comment:
this will start a new runtime. Better to take the `ConcurrencyConfig` from the outside.

Reply:
Is there a performance downside to starting a new runtime that does not actually do any work (unless we are getting invalid messages in which case performance is not our primary concern as something is broken)? I did it this way because I think it's simpler for the user to just not think about what config they need here and pass it to the stream processor, and it seems like it wouldn't impact perf under normal operation. Unlike the other examples where we share the config this isn't part of the user defined processing strategy.